src/mixins/java/org/spongepowered/common/mixin/core/world/level/ExplosionMixin.java

```diff
@@ -118,158 +113,107 @@ public abstract class ExplosionMixin implements ExplosionBridge {
         this.impl$knockback = 1.0;
     }
 
-    /**
-     * @author gabizou
-     * @author zidane
-     * @reason Fire ExplosionEvent.Detonate
-     */
-    @Overwrite
-    public void explode() {
-
-        // Sponge Start - Do not run calculation logic on client thread
+    @Inject(method = "explode", at = @At("HEAD"), cancellable = true)
+    public void impl$explode_clientCheck(CallbackInfo ci) {
         if (this.level.isClientSide) {
-            return;
+            ci.cancel();
```

Comment:
This could remain a return imo or, better, inject right after this check. I only propose this in-case Mojang does something different in the future and we forget about the cancel. Which effectively enters overwrite territory there.

Reply:
Maybe I misunderstand what you mean, however this inject seemed strange to me anyway. I only added it since the overwrite had it, but is there a particular reason why `explode` should only run on the server? The `explode` method does not have a client check itself. If the intention is to not fire the event on the client, would it be better to move this check next to the `ShouldFire.EXPLOSION_EVENT_DETONATE`?

Reply:
Part of the premise of not calling explode on the client is when Sponge is running on the client, it's expecting to only ever call this function on the server (which if you look at the usages of the injecting method, it is by default always called on the server game thread.